[ADD][16.0] stock_picking_restrict_cancel_printed

santostelmo opened the pull request:
Display printed flag on the form view of the transfer.
Raise an error when action cancel is called on a printed stock.picking: `You cannot cancel a picking that is already printed.``

santostelmo committed:
[ADD] stock_picking_restrict_cancel_printed

sebalix commented on stock_picking_restrict_cancel_printed/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,2 @@
+Display printed flag on the form view of the transfer.
+Raise an error when action cancel is called on a printed stock.picking: `You cannot cancel a picking that is already printed.``
```
```suggestion
Raise an error when canceling a printed stock transfer: `You cannot cancel a transfer that is already printed.`
```

santostelmo replied:
Thank you! Description updated

sebalix commented on stock_picking_restrict_cancel_printed/models/stock_picking.py:
```diff
@@ -0,0 +1,17 @@
+        for pick in self:
+            if pick.printed:
+                raise UserError(
+                    _("You cannot cancel a picking that is already printed.")
```
```suggestion
                    _("You cannot cancel a transfer that is already printed.")
```

rousseldenis commented:
@santostelmo @sebalix Could you look at https://github.com/OCA/stock-logistics-workflow/tree/16.0/stock_picking_start ?

As there are similar processes. @jbaudoux

jbaudoux commented:
> @santostelmo @sebalix Could you look at https://github.com/OCA/stock-logistics-workflow/tree/16.0/stock_picking_start ?
> 
> As there are similar processes. @jbaudoux

It's a bit different. Here we want to prevent cancelation if printed (really printed, not started yet)

cyrilmanuel reviewed (approved):
Not an expert But LGTM

jbaudoux commented on stock_picking_restrict_cancel_printed/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+    "development_status": "Beta",
+    "category": "Inventory",
+    "summary": "Prevent canceling a picking with a printed flag",
+    "author": "Camptocamp, Odoo Community Association (OCA)",
```
```suggestion
    "author": "Camptocamp, BCIM, Odoo Community Association (OCA)",
    "maintainers" : ["jbaudoux"],
```

jbaudoux commented on stock_picking_restrict_cancel_printed/readme/CONTRIBUTORS.rst:
```diff
@@ -0,0 +1 @@
+* Telmo Santos <telmo.santos@camptocamp.com>
```
```suggestion
* Telmo Santos <telmo.santos@camptocamp.com>
* Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
```

cyrilmanuel commented:
hi @rousseldenis, could you merge this PR please ?

rousseldenis commented:
@santostelmo @jbaudoux IMHO this deserves a configuration parameter on picking type side

sebalix commented on stock_picking_restrict_cancel_printed/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Stock picking restrict partner cancel printed",
```
```suggestion
    "name": "Stock Picking - Restrict cancellation if printed",
```

sebalix commented on stock_picking_restrict_cancel_printed/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+    "version": "16.0.1.0.0",
+    "development_status": "Beta",
+    "category": "Inventory",
+    "summary": "Prevent canceling a picking with a printed flag",
```
```suggestion
    "summary": "Prevent canceling a stock transfer if printed",
```

TDu committed:
Add option on stock_picking_restrict_cancel_printed

Add an option on the picking type, to disable the restriction
included by the module.

TDu commented:
> @santostelmo @jbaudoux IMHO this deserves a configuration parameter on picking type side

This commit https://github.com/OCA/stock-logistics-workflow/pull/1568/commits/a34ed2b793018cd5a8d5fbc84e5475888cb397f1 in the backport to v14 adds the option.

jbaudoux commented:
> > @santostelmo @jbaudoux IMHO this deserves a configuration parameter on picking type side
> 
> This commit [a34ed2b](https://github.com/OCA/stock-logistics-workflow/commit/a34ed2b793018cd5a8d5fbc84e5475888cb397f1) in the backport to v14 adds the option.

@TDu No PR with that commit on this PR? @santostelmo Will you cherry-pick it?

TDu commented:
Not sure why there is this message. But there is a commit in the backport that adds the option.

santostelmo commented:
@jbaudoux @TDu I cherry picked the commit with the configuration parameter

jbaudoux commented:
/ocabot merge  nobump
